checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,375 @@
+      for (AnnotationMirror co : createsObligations) {
+        JavaExpression expr = getCreatesObligationExpression(co, n, atypeFactory, supplier);
+        if (expr != null && !results.contains(expr)) {
+          results.add(expr);
+        }
+      }
+    }
+    AnnotationMirror createsObligation =
+        atypeFactory.getDeclAnnotation(n.getTarget().getMethod(), CreatesObligation.class);
+    if (createsObligation != null) {
+      JavaExpression expr =
+          getCreatesObligationExpression(createsObligation, n, atypeFactory, supplier);
+      if (expr != null && !results.contains(expr)) {
+        results.add(expr);
+      }
+    }
+    return results;
+  }
+
+  /**
+   * Parses a single CreatesObligation annotation. Clients should use {@link
+   * #getCreatesObligationExpressions(MethodInvocationNode, GenericAnnotatedTypeFactory,
+   * CreatesObligationElementSupplier)} instead.
+   *
+   * @param createsObligation a @CreatesObligation annotation
+   * @param n the invocation of a reset method
+   * @param atypeFactory the type factory
+   * @param supplier a type factory that can supply the executable elements for CreatesObligation
+   *     and CreatesObligation.List's value elements. Usually, you should just pass atypeFactory
+   *     again. The arguments are different so that the given type factory's adherence to both
+   *     protocols are checked by the type system.
+   * @return the Java expression representing the target, or null if the target is unparseable
+   */
+  private static @Nullable JavaExpression getCreatesObligationExpression(
+      AnnotationMirror createsObligation,
+      MethodInvocationNode n,
+      GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+      CreatesObligationElementSupplier supplier) {
+    // Unfortunately, there is no way to avoid passing the default string "this" here. The default
+    // must be hard-coded into the client, such as here. That is the price for the efficiency of not
+    // having to query the annotation definition (such queries are expensive).
+    String targetStrWithoutAdaptation =
+        AnnotationUtils.getElementValue(
+            createsObligation, supplier.getCreatesObligationValueElement(), String.class, "this");
+    // TODO: find a way to also check if the target is a known tempvar, and if so return that. That
+    // should improve the quality of the error messages we give.
+    JavaExpression targetExpr;
+    try {
+      targetExpr =
+          StringToJavaExpression.atMethodInvocation(
+              targetStrWithoutAdaptation, n, atypeFactory.getChecker());
+      if (targetExpr instanceof Unknown) {
+        issueUnparseableError(n, atypeFactory, targetStrWithoutAdaptation);
+        return null;
+      }
+    } catch (JavaExpressionParseException e) {
+      issueUnparseableError(n, atypeFactory, targetStrWithoutAdaptation);
+      return null;
+    }
+    return targetExpr;
+  }
+
+  /**
+   * Issues a mustcall.not.parseable error.
```

Comment:
This key "mustcall." is misleading since the error message is about a `@CreatesObligation` annotation.  I would rename it, or else give a better explanation here about what the method does.